okapi/services/caches/save_user_coords/docs.xml

```diff
@@ -0,0 +1,30 @@
+<xml>
+    <brief>Update personal coordinates of a geocache</brief>
+    <issue-id>629</issue-id>
+    <desc>
+        <p>This method allows your users to to change the content of the user's
+        personal cache coordinates from one value to the other value.</p>
+
+        <p>Current personal coordinates for the geocache can be retrieved
+        using the <b>alt_wpts</b> field in the
+        <a href="%OKAPI:methodargref:services/caches/geocache#fields%">services/caches/geocache</a>
+        method.</p>
```

Comment:
The docs for this field say that "this type of waypoint is available only in some installations". Please explain how this method will behave in case of installations which do not support personal coordinates.

Reply:
I have checked `opencaching.pl`, it has the same database table so the code will work. Whether `opencaching.pl` renders this detail at the UI, I don't know. Am I supposed to check all other installations too?

I don't know how I could verify the code changes on a different installation.

Reply:
You need to check only OCDE and OCPL only, all the other ones use the code from one of these. We can add reviewers from OCPL later on, or you can ask for them now (email rt@).

Reply:
Ok, I just reviewd the database layout for opencaching.pl and as it seems, if this is the right place: `https://opencaching.pl/okapi/devel/dbstruct` the OCPL does not have a `user_coord` field in the respective table, which has even a different name and different fields. Now the question what to do here? I could make the implementation conditional to OCDE and do nothing for OCPL but the whole method should then not even show up for the OCPL installation in which case the conditional implementation wouldn't be needed alltogether.

Could the service be conitionally added to the service runner such that it doesn't even show up on the list for the OCPL installation and also not on their documentation?

Reply:
You could add a method with `ocde-specific` infotag (see the docs for `services/apiref/method`). But let's first confirm with OCPL guys if indeed OCPL doesn't keep this data. It's possible that OCPL still has it, just in some other table/field. If that's the case, then we should support it.

@opencaching/opencaching-pl-lead-programmers Can you comment?

Reply:
I looked at the dbstruct for OCPL and the following seems to be the right place:

```
DROP TABLE IF EXISTS `cache_mod_cords`;
/*!40101 SET @saved_cs_client     = @@character_set_client */;
/*!40101 SET character_set_client = utf8 */;
CREATE TABLE `cache_mod_cords` (
  `cache_id` int(11) NOT NULL,
  `user_id` int(11) NOT NULL,
  `date` timestamp NOT NULL DEFAULT current_timestamp(),
  `longitude` double NOT NULL,
  `latitude` double NOT NULL,
  PRIMARY KEY (`cache_id`,`user_id`),
  KEY `user_id` (`user_id`)
) ENGINE=InnoDB DEFAULT CHARSET=utf8mb3 COLLATE=utf8mb3_general_ci COMMENT='Custom coordinates of geocache set by user';
/*!40101 SET character_set_client = @saved_cs_client */;
```
I will update the code accordingly, however, I do not have an environment where I could test that code path

Reply:
Remark: I have not found out conclusively if the OCPL installation makes the `user-coords` available in the `alt_wpts` dictionary for `services/caches/geocache`. The documentation says (as you stated before):

>user-coords - extra coordinates supplied by the user who had found the cache (NOT the owner of the cache), most probably pointing to the final location of the cache (e.g. a quiz cache); this type of waypoint is available only in some installations and only if you're using Level 3 Authentication,

Reply:
It might be that the documentation is outdated.

Reply:
> It might be that the documentation is outdated.

I am not exactly sure what documentation you mean. Is it this hier in `docs.xml` or somewhere else? If it is this file, the only thing I could think of is relatet to how the value (user_coords) could be retrieved in an OCPL environment? Ihave stated how it could be retrieved from OCDE but I don't actually know how id can be done in OCPL. It seems like you would know how that works in OCPL?